pkg/artifactory/repositories.go

```diff
@@ -1132,11 +1241,231 @@ func mkResourceSchema(skeema map[string]*schema.Schema, packer PackFunc, unpack
 			StateContext: schema.ImportStatePassthroughContext,
 		},
 
-		Schema: skeema,
+		Schema:        skeema,
 		CustomizeDiff: projectEnvironmentsDiff,
 	}
 }
 
+//Returns random string from a map[string]string
+func selectRandomFromMapOfStrings(m map[string]string) string {
+	mapLength := len(m)
+	allValues := make([]string, 0, mapLength)
+	for _, value := range m {
```

Comment:
How about you just loop through the map and randomly decide to select or not? Way simpler and no copying

```go
for _, value range m {
    if selectRand(true,false).(bool) {
     return value;
    }
}
```

Reply:
Or use `randBool()` 😄

Reply:
Not applicable any more